core/src/main/java/hudson/tools/ToolDescriptor.java

```diff
@@ -134,6 +136,7 @@ public void setInstallations(T... installations) {
      * Default value for {@link ToolInstallation#getProperties()} used in the form binding.
      * @since 1.305
      */
+    @Restricted(NoExternalUse.class)
```

Comment:
Add `@RestrictedSince("TODO")` below

Reply:
Would restricting this class prevent it from being overridden [here](https://github.com/jenkinsci/coverity-plugin/blob/8016930344f94474b5bd240ef1a033d497a89c65/src/main/java/jenkins/plugins/coverity/CoverityToolInstallation.java#L92)? There is also [this](https://github.com/jenkinsci/jenkins.py/blob/cbfc2ec231b447638d2d3e3bf82e3bc7bb160549/python-wrapper/src/main/java/jenkins/python/descriptor/ToolDescriptorPW.java#L198) use in jenkins.py but I don't think that is actually in use anywhere.

Reply:
as mentioned by batmat. Coverity is deprecatedÆ https://github.com/jenkinsci/jenkins/pull/3992#issuecomment-485018632

Reply:
```suggestion
    @Restricted(NoExternalUse.class)
    @RestrictedSince("2.177")
```